docs/integration_tests.md

```diff
@@ -0,0 +1,62 @@
+# Integration Tests
+
+Integration tests in Flutter allow self-driven end-to-end
```

Comment:
(continuing from https://github.com/zulip/zulip-flutter/pull/317#discussion_r1377596741 )

OK. Let's say "self-driving", then — that makes the car analogy more evident because that's the wording that's always used in that context, and that's also the wording they use in that `package:integration_test` readme.